### ISSUE TITLE
warning: useless cast to type ‘size_t’ {aka ‘unsigned int’} [-Wuseless-cast]

### ISSUE THREAD

**Issue by malaterre** [team-charls/charls#161]
Posting as issues for now:

* https://buildd.debian.org/status/fetch.php?pkg=charls&arch=i386&ver=2.3.1%2Bdfsg-1&stamp=1643190224&raw=0

```
/<<PKGBUILDDIR>>/src/lookup_table.h: In member function ‘void charls::golomb_code_table::add_entry(uint8_t, charls::golomb_code)’:
/<<PKGBUILDDIR>>/src/lookup_table.h:48:30: warning: useless cast to type ‘size_t’ {aka ‘unsigned int’} [-Wuseless-cast]
   48 |         for (size_t i{}; i < static_cast<size_t>(1U) << (byte_bit_count - length); ++i)
      |         
```

and

```
/<<PKGBUILDDIR>>/fuzztest/main.cpp: In function ‘int main(int, const char* const*)’:
/<<PKGBUILDDIR>>/fuzztest/main.cpp:75:68: warning: useless cast to type ‘unsigned int’ [-Wuseless-cast]
   75 |                 const ssize_t result{write(1, encoded_data.data(), static_cast<unsigned int>(encoded_data.size()))};
      |                                                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/<<PKGBUILDDIR>>/fuzztest/main.cpp:99:62: warning: useless cast to type ‘unsigned int’ [-Wuseless-cast]
   99 |         const size_t input_length = _read(fd, source.data(), static_cast<unsigned int>(source.size()));
      |                                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[ 68%] Linking CXX executable fuzztest
```

**Comment by malaterre** [comment]
oh neat trick:

* https://github.com/nlohmann/json/pull/2902/files#diff-3da1506d2888e4cd64eb9ae2874e721d0185ce73a4e74b92cff43f63b3a098b8

**Comment by vbaderks** [comment]
Using conditional_static_cast is indeed a very clean way to solve it.

**Comment by vbaderks** [comment]
This has been resolved with c15be902b01cbc85d539ac8e300148f6b43bad7a
